### PR TITLE
GNNE-1714:Fix/regression bugs

### PR DESCRIPTION
**Pull request by uranus0515**
1, Implement ConvTranspose to avoid OrtKi bugs;
2, Avoid null name caused by reference types;

### CODE REVIEW

File: src/Nncase.Evaluator/NN/Conv2DTranspose.cs [hunk 1]
```diff
@@ -27,24 +27,88 @@ public IValue Visit(IEvaluateContext context, Conv2DTranspose conv)
...
+
+        int inputIndex = 0;
+        for (int batch = 0; batch < inputShape[0]; batch++)
+        {
+            var out_batch_p = outCache.AsSpan().Slice(batch * (int)outputShape[1] * (int)outputShape[2] * (int)outputShape[3]);
+
+            for (int g = 0; g < groups; g++)
+            {
+                var out_group_p = out_batch_p.Slice(g * (int)g_oc * (int)outputShape[2] * (int)outputShape[3]);
+                var w_group_p = weights.ToArray<float>().AsSpan().Slice((int)g * (int)g_oc * (int)g_ic * (int)kernelShape[2] * (int)kernelShape[3]);
```
Comment (FusionBolt) [inline src/Nncase.Evaluator/NN/Conv2DTranspose.cs:53]: ToArray should be called outside the loop.

File: src/Nncase.Evaluator/NN/Conv2DTranspose.cs [hunk 2]
```diff
@@ -27,24 +27,88 @@ public IValue Visit(IEvaluateContext context, Conv2DTranspose conv)
...
+                            int filter_x_end = (int)System.Math.Min(kernelShape[3], ((int)outputShape[3] - out_x_origin + dilation[1] - 1) / dilation[1]);
+
+                            float in_v;
+                            if (ix < 0 || ix >= inputShape[3] || iy < 0 || iy >= inputShape[2])
+                            {
+                                in_v = 0f;
+                            }
+                            else
+                            {
+                                in_v = input.ToArray<float>()[inputIndex];
```
Comment (FusionBolt) [inline src/Nncase.Evaluator/NN/Conv2DTranspose.cs:75]: same as above

File: src/Nncase.Evaluator/NN/Conv2DTranspose.cs [hunk 3]
```diff
@@ -27,24 +27,88 @@ public IValue Visit(IEvaluateContext context, Conv2DTranspose conv)
...
+                        }
+                    }
+                }
+            }
+        }
+
+        for (int i = 0; i < outputSize; i++)
+        {
+            var biasIdx = i / (outputShape[2] * outputShape[3]) % outputShape[1];
+            outCache[i] = outCache[i] + bias.ToArray<float>()[biasIdx];
```
Comment (FusionBolt) [inline src/Nncase.Evaluator/NN/Conv2DTranspose.cs:108]: same as above

**Review by FusionBolt — CHANGES_REQUESTED** [review]
ToArray should be called outside the loop. Otherwise it will be very slow.

File: src/Nncase.Evaluator/NN/Conv2DTranspose.cs [hunk 4]
```diff
@@ -27,24 +27,88 @@ public IValue Visit(IEvaluateContext context, Conv2DTranspose conv)
...
-            outputShape,
-            pads,
-            stride).ToValue();
+        var inputShape = input.Shape;
+
+        var outputSize = outputShape[0] * outputShape[1] * outputShape[2] * outputShape[3];
+        float[] outCache = new float[outputSize];
+        Array.Clear(outCache, 0, (int)outputSize);
+
+        var g_ic = inputShape[1] / groups;
```
Comment (xhuohai) [inline src/Nncase.Evaluator/NN/Conv2DTranspose.cs:42]: camel case is better

**Review by xhuohai — CHANGES_REQUESTED** [review]

**Review by xhuohai — APPROVED** [review]

**Review by FusionBolt — APPROVED** [review]

**Pull request merged**